typed/remove.go

```diff
@@ -46,6 +48,7 @@ func (w *removingWalker) doList(t *schema.List) (errs ValidationErrors) {
 
 	// If list is null, empty, or atomic just return
 	if l == nil || l.Length() == 0 || t.ElementRelationship == schema.Atomic {
+		w.out = l
```

Comment:
This is it. `l` is a `listUnstructured` in your case. I'm a little concerned that this is not consistent with the map case, so we must have a bug there and a scenario that is mis-tested.

Reply:
This doesn't do the same thing, is this expected: https://github.com/kubernetes-sigs/structured-merge-diff/pull/133/files#diff-11232f9aff4ebc184b767f3a3520214aR80